ets_fiber_assigner/netflow.py

```diff
@@ -453,6 +462,11 @@ def buildProblem(bench, targets, tpos, classdict, tvisit, vis_cost=None,
                     dist = np.min(np.abs(closestDotsList[cidx]-tpos[ivis][tidx]))
                     tcost += blackDotPenalty(dist)
                 prob.cost += f*tcost
+        # If requested, penalize non-allocated fibers
+        if fiberNonAllocationCost != 0:
+            relevantVars = [var for (keys, var) in Cv_i.items() if keys[1] == ivis]
```

Comment:
This didn't work for me. The following modification works but I'm not sure if it is right.
```suggestion
            relevantVars = [var[0] for (keys, var) in Cv_i.items() if keys[1] == ivis]
```

Reply:
Thank you! I'm trying to reproduce this, but unfortunately I currently have unrelated failures:

```
  File "/home/martin/codes/efa/pfs/utils/coordinates/DistortionCoefficients.py", line 640, in radec_to_subaru
    tel2 = Observer.at_site("Subaru", timezone="US/Hawaii")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/astroplan/observer.py", line 281, in at_site
    return cls(location=EarthLocation.of_site(site_name), name=name, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/astropy/units/decorators.py", line 313, in wrapper
    return_ = wrapped_function(*func_args, **func_kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/astroplan/observer.py", line 207, in __init__
    self.timezone = pytz.timezone(timezone)
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/pytz/__init__.py", line 202, in timezone
    raise UnknownTimeZoneError(zone)
pytz.exceptions.UnknownTimeZoneError: 'US/Hawaii'
```

I hope I can fix this soon and then continue debugging the problem.

Reply:
I think I understand the problem: `relevantVars` was not (as I had expected) a list of LP variables, but a list of lists. The COIN solver I used accepted this without complaint, but Gurobi doesn't seem to like it. I'm now unpacking the list, which hopefully gets things working again.